internal/parse/parser_test.go

```diff
@@ -44,8 +42,8 @@ var parseTests = []parseHelperTest{
 
 func TestRunTable(t *testing.T) {
 	for _, v := range parseTests {
-		p.Parse(v.input)
-		for i, _ := range v.result {
+		p.init(v.input)
```

Comment:
Blackbox tests are generally much nicer than whitebox ones, as they test the real external expectations of the implementation, while allowing the internal implementation to change freely as long as the public promises are respected.  Sometimes we may need to dig into the implementation to test certain things, but it doesn't look to be the case here.  Can we please rename this package (first line) to "parser_test"?  This will make us more honest about not touching the internal implementation unless that's really required.  Not sure if we have other test files in a similar situation, but if so can we please change them too?

Reply:
Done. The parse function tests that are necessarily whitebox are now in parse.go and the rest in separate test packages.